### PR TITLE
Adds Remove Contacts Functionality

### LINKED ISSUE #1051: Remove a contact
Per user request, you should be able to delete a contact from your "address book". The API support this: `removeContact`

### PR DESCRIPTION
**Pull request by undercase**
Fixes #1051. Adds the ability to remove a contact from the `Contact` detail view/modal. To this end, I made a few other notable changes:

- The `Contact` view now accepts an `IContact` object as a parameter, instead of `avatar`, `username`, and `address`.

- There is now a button in the `Contact` view that will let the user add a contact (if it is not already in the user's contacts) or remove a contact (if it is).

- The Contact profile layout is now vertical, not horizontal, so that more functionality can be added without disrupting the layout.

### CODE REVIEW

File: App/Containers/Contact.tsx [hunk 1]
```diff
@@ -67,29 +80,59 @@ class ContactModal extends React.Component<Props> {
...
+    isContact,
+    removing,
+    adding
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<RootAction>): DispatchProps => {
+  return {
+    removeContact: (address: string) => dispatch(contactsActions.removeContactRequest(address)),
+    addContact: (contact: IContact) => dispatch(contactsActions.addContactRequest(contact))
```
Comment (asutula) [inline App/Containers/Contact.tsx]: One small, but nice improvement is using the optional `ownProps` parameter in `mapDispatchToProps`. Then you don't have to have `address: string` and `contact: IContact` as parameters to these prop functions... Inside the functions, you can just pull `address` and `contact` from `ownProps`.

File: App/features/contacts/actions.ts [hunk 2]
```diff
@@ -50,6 +50,22 @@ export const clearAddContact = createAction('@contacts/CLEAR_ADD_CONTACT_ERROR',
...
+  return (address: string) => resolve({ address })
+})
+
+export const removeContactSuccess = createAction('@contacts/REMOVE_CONTACT_SUCCESS', (resolve) => {
+  return (address: string) => resolve ({ address })
+})
+
+export const removeContactError = createAction('@contacts/REMOVE_CONTACT_ERROR', (resolve) => {
+  return (address: string, error: any) => resolve({ address, error })
+})
```
Comment (asutula) [inline App/features/contacts/actions.ts]: It's good to use `typesafe-actions`' `createAsyncAction` for these request, success, error setups. It's cleaner. You can find examples in other places in our codebase.

File: App/features/contacts/actions.ts [hunk 3]
```diff
@@ -50,6 +50,22 @@ export const clearAddContact = createAction('@contacts/CLEAR_ADD_CONTACT_ERROR',
...
+
+export const clearRemoveContactError = createAction('@contacts/CLEAR_REMOVE_CONTACT_ERROR', (resolve) => {
```
Comment (asutula) [inline App/features/contacts/actions.ts]: Let's not use the `@` prefix on those strings, not sure how/why I started that.

**Comment by undercase** [comment]
Fixed the issues you brought up. Thanks 👍.

File: App/Containers/Contact.tsx [hunk 4]
```diff
@@ -1,36 +1,46 @@
...
   threadThumbs: ReadonlyArray<ThreadThumbs>
+  isContact: boolean
+  removing: boolean
+  adding: boolean
 }
 
-type Props = StateProps & NavigationScreenProps<NavProps>
+interface DispatchProps {
+  removeContact: (address: string) => void
+  addContact: (contact: IContact) => void
```
Comment (asutula) [inline App/Containers/Contact.tsx]: So you still need to remove the arguments from these.

File: App/Containers/Contact.tsx [hunk 5]
```diff
@@ -67,29 +80,61 @@ class ContactModal extends React.Component<Props> {
...
     )
   }
+
+  onRemove = () => {
+    this.props.removeContact(this.props.navigation.getParam('contact').address)
+  }
+
+  onAdd = () => {
+    this.props.addContact(this.props.navigation.getParam('contact'))
+  }
```
Comment (asutula) [inline App/Containers/Contact.tsx]: And same in the calls inside of these

Comment (undercase) [reply App/Containers/Contact.tsx]: Oof idk how I missed this. Fixing it right now.

**Pull request merged**